### ISSUE TITLE
Dropout implementation ignores manual_seed, produces non-reproducible results

### ISSUE THREAD

**Issue by lvyufeng** [FlagOpen/FlagGems#6217]
## Bug Description

The FlagGems dropout implementation does not respect PyTorch's manual seeding, producing non-reproducible results even when called with identical seeds.

## Reproduction

```python
import torch
torch.manual_seed(1234)
result1 = torch.nn.functional.dropout(torch.ones(256, device='cuda'), p=0.5, training=True)

torch.manual_seed(1234)
result2 = torch.nn.functional.dropout(torch.ones(256, device='cuda'), p=0.5, training=True)

assert torch.equal(result1, result2)  # FAILS with FlagGems
```

## Expected Behavior

When the same manual seed is set before calling dropout, the function should produce identical dropout masks. This is the behavior with PyTorch's native CUDA dropout implementation.

## Actual Behavior

FlagGems dropout produces different masks on each call, even with identical seeds:
- First call with seed=1234: `tensor([2., 0., 0., 2., 0., 2., 2., 2., ...])`
- Second call with seed=1234: `tensor([0., 0., 0., 2., 2., 2., 2., 0., ...])`

## Environment

- PyTorch: 2.4.0+
- CUDA: 12.1
- FlagGems: commit 15be61e6eb64c29b63e137850523f4be11f2e70c
- Platform: CUDA, also affects PPU

## Impact

This breaks:
1. Reproducible training runs
2. Debugging workflows that depend on deterministic behavior
3. Testing frameworks that verify dropout behavior

## Workaround

Fall back to PyTorch's native CUDA dropout implementation by not routing dropout operators to FlagGems.

## Related Test

The issue was discovered in our CI test suite:
- Test: `test_dropout_reproducible_on_flaggems_path` in `tests/integration/ops/test_rng_dispatch.py`
- Expected: Same seed produces same dropout mask
- Actual: Each call produces different mask
